content/tutorials/reactor-firehose-kinesis.textile

```diff
@@ -0,0 +1,602 @@
+
+Create the @views/index.html@ file. This HTML page contains a button which you will click to generate random sensor data to send to your data stream. It will also ultimately display the average values of those sensor readings:
+
+```[html]
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>AWS Kinesis Firehose Demo</title>
+    <link rel="stylesheet" href="style.css" />
+    <script src="https://cdn.ably.io/lib/ably.min-1.js"></script>
+    <script src="client.js"></script>
+  </head>
+  <body>
+    <h1>AWS Kinesis Firehose Demo</h1>
+    <div>
+      <button id="generateSensorData" onclick="generateSensorData()">
+        Generate Readings
+      </button>
+    </div>
+    <div>
+      <p id="count"></p>
+      <p id="averages"></p>
+    </div>
+  </body>
+</html>
+```
+
+Create the @public/style.css@ stylesheet to include some basic styling for the page:
+
+```[css]
+body {
+  background-color: skyblue;
+  padding-left: 10px;
+  font-size: 14pt;
+  font-family: Verdana, Geneva, Arial, Helvetica, sans-serif;
+  color: black;
+}
+
+h1 {
+  font: 20pt Verdana, Geneva, Arial, Helvetica, sans-serif;
+  font-weight: bold;
+  line-height: 20pt;
+}
+```
+
+Create the @public/client.js@ file. This will contain the client side code for displaying the sensor data averages. Leave it blank for now: you will write this code in a later step.
+
+Enter the following code in @server.js@:
+
+```[javascript]
+// server.js
+
+require("dotenv").config()
+
+const express = require("express")
+const rn = require("random-number")
+const Ably = require("ably")
+const ably = new Ably.Rest({ key: process.env.ABLY_API_KEY })
+
+const app = express()
+
+// make all the files in 'public' available
+app.use(express.static("public"))
+
+// load the home page, index.html
+app.get("/", (request, response) => {
+  response.sendFile(__dirname + "/views/index.html")
+})
+
+// listen for requests
+const listener = app.listen(process.env.PORT, () => {
+  console.log("Your app is listening on port " + process.env.PORT)
+})
+```
+
+This code imports all the required dependencies, instantiates the "Ably REST API client":/documentation/rest and serves the @index.html@ web page and supporting files. Test it by running @node server.js@ and visiting @http://localhost:5000@ in your browser. If everything is working correctly, you should see a web page with a button inviting you to generate sensor readings. The button doesn't do anything yet: you will write the code for this in the next step. 
+
+h2. Step 5: Generate random sensor data
+
+In this step, you will use the Node.js @random-number@ module to generate random sensor readings that contain temperature and humidity values and then publish them to the @raw-sensor-data@ channel. Because you created a Reactor Firehose integration rule on that channel, the transactions you created will be sent to your data stream.
```

Comment:
```suggestion
In this step, you will use the Node.js @random-number@ module to generate random sensor readings that contain temperature and humidity values and then publish them to the @raw-sensor-data@ channel. Because you created a Reactor Firehose integration rule on that channel, the transactions you create will be sent to your data stream.
```
```suggestion
In this step, you will use the Node.js @random-number@ module to generate random sensor readings that contain temperature and humidity values and then publish them to the @raw-sensor-data@ channel. Because you created a Reactor Firehose integration rule on that channel, the transactions you created will be sent to your data stream.
```